File: app/src/main/java/com/samhalperin/cooperhewitt/masterscreen/MasterContract.java
package com.samhalperin.cooperhewitt.masterscreen;

import com.samhalperin.cooperhewitt.data.models.searchobjects.SearchObject;
import com.samhalperin.cooperhewitt.data.models.searchobjects.SearchObjects;
import com.samhalperin.cooperhewitt.data.repository.RepositoryContract;

import java.util.List;

public interface MasterContract {

    public interface View {
        void navigateToAboutActivity();
        void navigateToDetailActivity(String id);
        void displayPage(List<SearchObject> items);
        void displayProgressBar();
        void hideProgressBar();
        void clearResults();

    }

    public interface UserActionHandler {
        void loadPage(int pageNumber, boolean clear);
        void selectArtisticPeriod(int period);
        void navigateToDetailActivity(String id);
        void navigateToAboutActivity();
    }
}

File: app/src/main/java/com/samhalperin/cooperhewitt/masterscreen/MasterContract.kt
package com.samhalperin.cooperhewitt.masterscreen

import com.samhalperin.cooperhewitt.data.models.searchobjects.SearchObject
import com.samhalperin.cooperhewitt.data.models.searchobjects.SearchObjects
import com.samhalperin.cooperhewitt.data.repository.RepositoryContract

interface MasterContract {

    interface View {
        fun navigateToAboutActivity()
        fun navigateToDetailActivity(id: String)
        fun displayPage(items: List<SearchObject>)
        fun displayProgressBar()
        fun hideProgressBar()
        fun clearResults()

    }

    interface UserActionHandler {
        fun loadPage(pageNumber: Int, clear: Boolean)
        fun selectArtisticPeriod(period: Int)
        fun navigateToDetailActivity(id: String)
        fun navigateToAboutActivity()
    }
}
